### PR TITLE
Controlgroup fixes

### PR DESCRIPTION
**Pull request by arschmitz**
Fixes various issues with controlgroups

### CODE REVIEW

File: ui/widgets/controlgroup.js [hunk 1]
```diff
@@ -180,22 +186,31 @@ return $.widget( "ui.controlgroup", {
...
 
 			}[ position ]
 		};
 	},
 
+	_resolveClassesValues: function( classes, instance ) {
+		$.each( classes, function( key ) {
+			var current = instance.options.classes[ key ] || "";
+			current = current.replace( removeClassRegex, "" ).trim();
+			classes[ key ] = ( current + " " + classes[ key ] ).replace( / +/g, " " );
```
Comment (scottgonzalez) [inline ui/widgets/controlgroup.js]: Can you switch the space literal to `\s` in the regex?

File: ui/widgets/controlgroup.js [hunk 2]
```diff
@@ -30,6 +30,7 @@
 		factory( jQuery );
 	}
 }( function( $ ) {
+var removeClassRegex = /ui-corner-([a-z]){2,6}/g;
```
Comment (scottgonzalez) [inline ui/widgets/controlgroup.js]: If this stays out of the prototype, then it needs to be prefixed with `controlgroup`. The factory goes away during builds and all variables get put in the same scope.

File: tests/unit/controlgroup/core.js [hunk 3]
```diff
@@ -74,4 +74,29 @@ QUnit.test( "selectmenu: controlgroupLabel", function( assert ) {
...
+}
+
+QUnit.test( "_resolveClassesValues", function( assert ) {
+	assert.expect( 6 );
+	assertSanatized( assert, "bar ui-corner-bottom", "bar", "Single Corner Class Removed end" );
+	assertSanatized( assert, "ui-corner-bottom bar", "bar", "Single Corner Class Removed beginning" );
+	assertSanatized( assert, "bar ui-corner-bottom ui-corner-left", "bar", "Multiple Corner Class Removed end" );
+	assertSanatized( assert, "ui-corner-bottom ui-corner-left bar", "bar", "Multiple Corner Class Removed beginning" );
+	assertSanatized( assert, "bar ui-corner-bottom ui-corner-left foo", "bar foo", "Multiple Corner Class Removed Middle" );
+	assertSanatized( assert, "bar", "bar", "No corner Class" );
```
Comment (scottgonzalez) [inline tests/unit/controlgroup/core.js]: Update the casing in the messages so only the first word is capitalized. Fix plurality of class in "multiple corner class".

File: ui/widgets/controlgroup.js (hunk 2 above)
Comment (arschmitz) [inline ui/widgets/controlgroup.js]: Oh right ill fix that

File: ui/widgets/controlgroup.js [hunk 4]
```diff
@@ -112,6 +112,11 @@ return $.widget( "ui.controlgroup", {
 				.each( function() {
 					var element = $( this );
 					var instance = element[ widget ]( "instance" );
+
+					// If the button is the child of a spinner ignore it
+					if ( widget === "button" && element.parent( ".ui-spinner" ).length ) {
```
Comment (scottgonzalez) [inline ui/widgets/controlgroup.js]: Seems like this should really be more generic. Something along the lines of if the found widget is inside any other widget which is also inside this controlgroup, then ignore it. The problem is the "which is also inside any other widget" part. I'm not sure there's anything we can do about this right now, but I really don't like that we're writing logic around specific widgets.

File: ui/widgets/controlgroup.js [hunk 5]
```diff
@@ -87,8 +87,11 @@ return $.widget( "ui.controlgroup", {
 			if ( widget === "controlgroupLabel" ) {
 				labels = that.element.find( selector );
 				labels.each( function() {
-					$( this ).contents()
-						.wrapAll( "<span class='ui-controlgroup-label-contents'></span>" );
+					var element = $( this );
+					if ( !element.children( ".ui-controlgroup-label-contents" ).length ) {
```
Comment (scottgonzalez) [inline ui/widgets/controlgroup.js]: You can invert this and return to avoid nested even deeper.

File: ui/widgets/controlgroup.js (hunk 4 above)
Comment (arschmitz) [inline ui/widgets/controlgroup.js]: I agree I just was not sure how else to do this unless we force enhancment within a controlgroup to be more explicit ala JQM data-role

**Comment by scottgonzalez** [comment]
Looks good.

**Pull request closed without merging**